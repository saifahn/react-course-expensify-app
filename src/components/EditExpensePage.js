import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }

  onClick = () => {
    this.props.removeExpense(this.props.expense.id)
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <h4>
          Editing the expense with id of
          {this.props.expense.id}
        </h4>
        <button onClick={this.onClick}>Remove</button>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: id => dispatch(removeExpense({ id })),
})

// props.match.params is the matched URL parameters
// find the expense which has the same ID as the ID of this page
// pass the expense down to EditExpensePage to be accessed in props
const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id
    }),
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(EditExpensePage)
