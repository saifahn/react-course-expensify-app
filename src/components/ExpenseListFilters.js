import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      calendarFocused: null,
    };
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    }
    if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          startDateId="startDateId"
          endDateId="endDateId"
          // onDatesChange={({ startDate, endDate }) => this.setState({ startDate, endDate })}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates
          numberOfMonths={1}
          isOutsideRange={() => false}
          // onFocusChange={focusedInput => this.setState({ focusedInput })}
        />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDate: endDate => dispatch(setEndDate(endDate)),
});

const mapStateToProps = state => (
  {
    filters: state.filters,
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
