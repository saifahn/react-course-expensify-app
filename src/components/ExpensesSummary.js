import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary = props => (
  <div>
    {props.expenseCount > 0
      &&
      <p>Viewing {props.expenseCount} expense{props.expenseCount > 1
        && 's'}, totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
      </p>}
  </div>
);

const mapStateToProps = (state) => {
  const expenses = getVisibleExpenses(state.expenses, state.filters);
  const expensesTotal = getExpensesTotal(expenses);
  return {
    expenseCount: expenses.length,
    expensesTotal,
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
