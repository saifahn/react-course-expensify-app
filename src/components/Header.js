import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <div className="header__links">
      <NavLink to="/" activeClassName="is-active" exact>Dashboard</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    </div>
  </header>
);

export default Header;
