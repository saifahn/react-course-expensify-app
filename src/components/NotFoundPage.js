import React from 'react'
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
  <div>
    404 - The page you were looking for {"isn't"} here!{' '}
    <Link to="/">Go home.</Link>
  </div>
)

export default NotFoundPage
