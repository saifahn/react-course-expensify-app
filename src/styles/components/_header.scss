.header__links {
  display: flex;
  // justify-content: space-between;
}

.is-active {
  font-weight: bold;
}