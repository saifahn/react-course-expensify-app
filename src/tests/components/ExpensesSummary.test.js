/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

// test('should render ExpensesSummary correctly on ExpenseDashboardPage', () => {
//   const wrapper = shallow(<ExpenseDashboardPage />);
//   expect(wrapper).toMatchSnapshot();
// });

test('should render ExpensesSummary correctly with more than one expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal="9434" />,
  )
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with one expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal="9434" />,
  )
  expect(wrapper).toMatchSnapshot()
})
