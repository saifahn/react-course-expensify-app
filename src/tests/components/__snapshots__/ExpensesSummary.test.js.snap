// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with more than one expense 1`] = `
<div>
  <p>
    Viewing 
    2
     expense
    s
    , totalling 
    $94.34
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
    , totalling 
    $94.34
  </p>
</div>
`;
